# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/libplacebo/hotfix-python.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

vulkan/utils_gen: fix for python 3.14
Python 3.14+ has added more type checking. This patch fixes usage

Fixes: #335

Signed-off-by: Nicolas Chauvet <kwizart@gmail.com>

https://github.com/haasn/libplacebo/commit/12509c0f1ee8c22ae163017f0a5e7b8a9d983a17

--- libplacebo-v7.351.0/src/vulkan/utils_gen.py.vanilla
+++ libplacebo-v7.351.0/src/vulkan/utils_gen.py
@@ -202,7 +202,8 @@
     if not xmlfile or xmlfile == '':
         xmlfile = find_registry_xml(datadir)
 
-    registry = VkXML(ET.parse(xmlfile))
+    tree = ET.parse(xmlfile)
+    registry = VkXML(tree.getroot())
     with open(outfile, 'w') as f:
         f.write(TEMPLATE.render(
             vkresults = get_vkenum(registry, 'VkResult'),
